### ISSUE TITLE
Sik-Ho Tang | Review -- RotNet: Unsupervised Representation Learning by Predicting Image Rotations.

### ISSUE THREAD

**Issue by NorbertZheng** [NorbertZheng/read-papers#131]
Sik-Ho Tang. [Review — RotNet: Unsupervised Representation Learning by Predicting Image Rotations](https://sh-tsang.medium.com/review-rotnet-unsupervised-representation-learning-by-predicting-image-rotations-60f4e4f3cf67).

**Comment by NorbertZheng** [comment]
## Overview
RotNet: Self-Supervised Learning by Predicting Image Rotations.

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/f69ce375-94a0-4f06-9621-551504a2b245)
The core intuition is that if someone is not aware of the concepts of the objects depicted in the images, he/she cannot recognize the rotation applied to the images.

In this story, Unsupervised Representation Learning by Predicting Image Rotations, by University Paris-Est, is reviewed. In this paper:
 - Using RotNet, image features are learnt by training ConvNets to recognize the 2d rotation that is applied to the image that it gets as input.
 - By this mean, unsupervised pre-trained [AlexNet](https://medium.com/coinmonks/paper-review-of-alexnet-caffenet-winner-in-ilsvrc-2012-image-classification-b93598314160) model achieves the state-of-the-art mAP of 54.4% that is **only 2.4 points lower from the supervised [AlexNet](https://medium.com/coinmonks/paper-review-of-alexnet-caffenet-winner-in-ilsvrc-2012-image-classification-b93598314160).**

This is a paper in 2018 ICLR with over 1100 citations.

**Comment by NorbertZheng** [comment]
## Image Rotation Prediction Framework
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/44279074-d299-48bc-b8be-86cff66ad5f7)
RotNet: Illustration of the self-supervised task.

Model:
 - Given four possible geometric transformations, the 0, 90, 180, and 270 degrees rotations, a ConvNet model $F(:)$ is trained to recognize the rotation that is applied to the image that it gets as input.
 - $F^{y}(X^{y})$ is the probability of rotation transformation $y$ predicted by model $F(:)$ when it gets as input an image that has been transformed by the rotation transformation $y$.

To successfully predict the rotation of an image, the ConvNet model must necessarily learn to localize salient objects in the image, recognize their orientation and object type, and then relate the object orientation with the dominant orientation that each type of object tends to be depicted within the available images.

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/f0a9c9b6-699a-4b29-bb30-9ca352aef7a8)
Attention maps generated by an [AlexNet](https://medium.com/coinmonks/paper-review-of-alexnet-caffenet-winner-in-ilsvrc-2012-image-classification-b93598314160) model trained (a) to recognize objects (supervised), and (b) to recognize image rotations (self-supervised).
The above attention maps are computed based on the magnitude of activations at each spatial cell of a convolutional layer and essentially reflect where the network puts most of its focus in order to classify an input image.
 - **Both supervised and self-supervised models seem to focus on roughly the same image regions.**

**Comment by NorbertZheng** [comment]
## Ablation Study & SOTA Comparison on CIFAR-10
### Supervised Training on Each Layer on CIFAR
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/6cbba403-34be-4846-b829-c6a8286e14ac)
Evaluation of the unsupervised learned features by measuring the classification accuracy that they achieve when training a non-linear object classifier on top of them.

Model:
 - The RotNet models with Network-In-Network ([NIN](https://towardsdatascience.com/review-nin-network-in-network-image-classification-69e271e499ee)) architectures are used. The total number of conv. layers of the examined RotNet models is 9, 12, and 15 for 3, 4, and 5 conv. blocks respectively.
 - The classifiers are learnt on top of conv layers. Those classifiers are trained in a supervised way on CIFAR-10. They consist of 3 fully connected layers; the 2 hidden layers have 200 feature channels each and are followed by batch-norm and ReLU units.
 - [AlexNet](https://medium.com/coinmonks/paper-review-of-alexnet-caffenet-winner-in-ilsvrc-2012-image-classification-b93598314160) or [NIN](https://towardsdatascience.com/review-nin-network-in-network-image-classification-69e271e499ee) is used as the ConvNet depending on the tasks.

Results:
 - In all cases, the feature maps generated by the 2nd conv. block (that actually has depth 6 in terms of the total number of conv. layer till that point) achieve the highest accuracy, i.e., between 88.26% and 89.06%.
 - Increasing the total depth of the RotNet models leads to increased object recognition performance by the feature maps generated by earlier layers.

**Comment by NorbertZheng** [comment]
### Number of Rotations
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/a105fe21-6bf3-428f-9ce4-d4e1163c08d5)
Exploring the quality of the self-supervised learned features w.r.t. the number of recognized rotations.

Results:
 - The 2 orientations case offers too few classes for recognition.
 - The 8 orientations case the geometric transformations are not distinguishable enough and furthermore the 4 extra rotations introduced may lead to visual artifacts on the rotated images.

The best one is the one using 4 rotations of 0, 90, 180, and 270.

**Comment by NorbertZheng** [comment]
### SOTA Comparison
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/df9071e9-ee2c-4618-a397-61c0ffe933d6)
Evaluation of unsupervised feature learning methods on CIFAR-10.

RotNet improves over the prior unsupervised approaches such as [Exemplar-CNN](https://sh-tsang.medium.com/review-exemplar-cnn-discriminative-unsupervised-feature-learning-with-convolutional-neural-fa68abe937cc) and [DCGAN](https://medium.com/@sh.tsang/review-dcgan-deep-convolutional-generative-adversarial-network-gan-ec390cded63c), and achieve state-of-the-art results in CIFAR-10.

More notably, the accuracy gap between the RotNet based model and the fully supervised [NIN](https://towardsdatascience.com/review-nin-network-in-network-image-classification-69e271e499ee) model is very small, only 1.64 percentage points (92.80% vs 91.16%).

**Comment by NorbertZheng** [comment]
## Task Generalization on ImageNet, Places, & PASCAL VOC
### ImageNet
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/8010cb5e-0335-48c7-b170-205346bd1162)
Task Generalization: ImageNet top-1 classification with non-linear layers.

RotNet surpasses all the other methods, such as [Context Prediction](https://sh-tsang.medium.com/review-unsupervised-visual-representation-learning-by-context-prediction-self-supervised-51a1d7ce6aff), [Colorization](https://sh-tsang.medium.com/review-colorful-image-colorization-self-supervised-learning-c2667640bd0a), [Jigsaw Puzzles](https://sh-tsang.medium.com/review-unsupervised-learning-of-visual-representations-by-solving-jigsaw-puzzles-50b14d755004), and [BiGAN](https://sh-tsang.medium.com/review-bigan-adversarial-feature-learning-gan-535eb76be2ca), by a significant margin.

![image](https://github.com/NorbertZheng/read-papers/assets/37235023/bae0ddb0-58e2-4f49-8f76-9fbb0ce58a7f)
Task Generalization: ImageNet top-1 classification with linear layers.

Model:
 - Regression classifiers on top of the feature maps of each layer are trained to perform the 1000-way ImageNet classification task.
 - All weights are frozen and feature maps are spatially resized (with adaptive max pooling) so as to have around 9000 elements.
 - All approaches use [AlexNet](https://medium.com/coinmonks/paper-review-of-alexnet-caffenet-winner-in-ilsvrc-2012-image-classification-b93598314160) variants and were pre-trained on ImageNet without labels except “ImageNet labels”.

Results:
 - RotNet demonstrates significant improvements over prior unsupervised methods, such as [Context Prediction](https://sh-tsang.medium.com/review-unsupervised-visual-representation-learning-by-context-prediction-self-supervised-51a1d7ce6aff), [Context Encoders](https://sh-tsang.medium.com/review-context-encoders-feature-learning-by-inpainting-bd181e48997), [Colorization](https://sh-tsang.medium.com/review-colorful-image-colorization-self-supervised-learning-c2667640bd0a), [Jigsaw Puzzles](https://sh-tsang.medium.com/review-unsupervised-learning-of-visual-representations-by-solving-jigsaw-puzzles-50b14d755004), [Split-Brain Auto](https://sh-tsang.medium.com/review-split-brain-autoencoders-unsupervised-learning-by-cross-channel-prediction-cebeba644a3c), and [BiGAN](https://sh-tsang.medium.com/review-bigan-adversarial-feature-learning-gan-535eb76be2ca).

**Comment by NorbertZheng** [comment]
### Places
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/000f0129-8c09-4a48-a915-b4b592814945)
Task & Dataset Generalization: Places top-1 classification with linear layers.

Similarly, RotNet manages to either surpass or achieve comparable results w.r.t. prior state-of-the-art unsupervised learning approaches, , such as [Context Prediction](https://sh-tsang.medium.com/review-unsupervised-visual-representation-learning-by-context-prediction-self-supervised-51a1d7ce6aff), [Context Encoders](https://sh-tsang.medium.com/review-context-encoders-feature-learning-by-inpainting-bd181e48997), [Colorization](https://sh-tsang.medium.com/review-colorful-image-colorization-self-supervised-learning-c2667640bd0a), [Jigsaw Puzzles](https://sh-tsang.medium.com/review-unsupervised-learning-of-visual-representations-by-solving-jigsaw-puzzles-50b14d755004), [Split-Brain Auto](https://sh-tsang.medium.com/review-split-brain-autoencoders-unsupervised-learning-by-cross-channel-prediction-cebeba644a3c), and [BiGAN](https://sh-tsang.medium.com/review-bigan-adversarial-feature-learning-gan-535eb76be2ca).

**Comment by NorbertZheng** [comment]
### PASCAL VOC
![image](https://github.com/NorbertZheng/read-papers/assets/37235023/a75e5d47-3ad8-4975-904b-e4ea977d6d8e)
Task & Dataset Generalization: PASCAL VOC 2007 classification and detection results, and PASCAL VOC 2012 segmentation results.

Model:
 - For classification, the features are either fixed before conv5 (column fc6–8) or the whole model is fine-tuned (column all). For detection multi-scale training and single scale testing are used. All approaches use [AlexNet](https://medium.com/coinmonks/paper-review-of-alexnet-caffenet-winner-in-ilsvrc-2012-image-classification-b93598314160) variants and were pre-trained on ImageNet without labels.

Results:
 - RotNet outperforms by significant margin all the competing unsupervised methods, such as [Context Prediction](https://sh-tsang.medium.com/review-unsupervised-visual-representation-learning-by-context-prediction-self-supervised-51a1d7ce6aff), [Context Encoders](https://sh-tsang.medium.com/review-context-encoders-feature-learning-by-inpainting-bd181e48997), [Colorization](https://sh-tsang.medium.com/review-colorful-image-colorization-self-supervised-learning-c2667640bd0a), [Jigsaw Puzzles](https://sh-tsang.medium.com/review-unsupervised-learning-of-visual-representations-by-solving-jigsaw-puzzles-50b14d755004), [Split-Brain Auto](https://sh-tsang.medium.com/review-split-brain-autoencoders-unsupervised-learning-by-cross-channel-prediction-cebeba644a3c), and [BiGAN](https://sh-tsang.medium.com/review-bigan-adversarial-feature-learning-gan-535eb76be2ca), in all tested tasks, significantly narrowing the gap with the supervised case.
 - Notably, the PASCAL VOC 2007 object detection performance that our self-supervised model achieves is 54.4% mAP, which is only 2.4 points lower than the supervised case.

**Comment by NorbertZheng** [comment]
## Reference
 - [2018 ICLR] [RotNet/Image Rotations] [Unsupervised Representation Learning by Predicting Image Rotations](https://arxiv.org/abs/1803.07728).
